front/src/common/authorization/utils/generateGrantSelectProps.ts

```diff
@@ -0,0 +1,75 @@
+import type { TFunction } from 'i18next';
+
+import type { Grant, PrivilegesByGrant } from 'common/api/mock/mockEditoastApi';
+import checkPrivileges from 'common/authorization/utils/checkPrivileges';
+
+import { GRANTS_LABEL } from '../consts';
+
+const generateGrantSelectProps = ({
+  privilegesByGrant,
+  subjectGrant,
+  connectedUserGrant,
+  t,
+}: {
+  privilegesByGrant?: PrivilegesByGrant | undefined;
+  subjectGrant?: Grant;
+  connectedUserGrant: Grant;
+  t: TFunction;
```

Comment:
I think you can specify here the translation namespace (it's for the script which checks the translation keys)